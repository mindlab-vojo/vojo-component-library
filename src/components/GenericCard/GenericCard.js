import React from 'react'
import PropTypes from 'prop-types'

import * as GenericCard from './GenericCard.style'

import ThemeConsumer from '../../style/ThemeManager/ThemeProvider'
import { Card } from '../Card/Card.style'
import Typography from '../Typography'
import IconManager from '../IconManager'

const CourseCardComponent = ({
  backgroundColor,
  borderRadius,
  children,
  fontWeightSubtitle,
  id,
  imageUrl,
  isFree,
  isNew,
  justifyContent,
  onClickCard,
  subtitle,
  subtitleColor,
  title,
  titleColor,
  titleSize,
  workload
}) => {

  const renderComponent = (theme) => (
    <Card
      backgroundColor={backgroundColor}
      hasPointer
      id={`GenericCard-${id}`}
      padding="15px 8px"
      onClick={onClickCard}>
      {isNew && <GenericCard.NewTag theme={theme}>
        <Typography
          color={theme.colors.whiteColor}>
          novidade
        </Typography>
      </GenericCard.NewTag>}
      <GenericCard.Wrapper
        className="GenericCard">
        <GenericCard.ImageBlock
          className="CourseCard__ImageBlock">
          <GenericCard.ImageWrapper
            className="CourseCard__ImageWrapper">
            <GenericCard.ImageCropper
              borderRadius={borderRadius}
              className="CourseCard__ImageWrapper__Cropper">
              <GenericCard.Image
                src={imageUrl}
                alt={title}
              />
            </GenericCard.ImageCropper>
          </GenericCard.ImageWrapper>
        </GenericCard.ImageBlock>
        <GenericCard.DataBlock justifyContent={justifyContent}>
          <GenericCard.Title>
            <Typography
              color={titleColor}
              fontSize={titleSize}
              fontWeight="bolder"
              fontFamily={"'Montserrat', sans-serif"}>
              {title}
            </Typography>
          </GenericCard.Title>
          {subtitle && <GenericCard.Subtitle>
            <Typography
              fontSize="12px"
              fontWeight={fontWeightSubtitle}
              color={subtitleColor ? subtitleColor : theme.colors.lightGreyColor}
              fontFamily={"'Rubik', sans-serif"}>
              {`${subtitle}`}
            </Typography>
          </GenericCard.Subtitle>}
          {isFree && <GenericCard.Free>
            <Typography
              fontSize="12px"
              fontWeight="bolder"
              color={'#666666'}
              fontFamily={"'Rubik', sans-serif"}
            >
              {"Gratuito"}
            </Typography>
          </GenericCard.Free>}
          {workload && <GenericCard.Workload>
            <IconManager icon={'Clock'} fill={'#666666'} width={'20px'} height={'20px'} />
            <GenericCard.WorkloadText>
              <GenericCard.WorkloadTextWrapper>
                <Typography
                  fontSize="12px"
                  color={theme.colors.darkGreyColor}
                  fontFamily={"'Rubik', sans-serif"}
                >
                  {"Carga hor√°ria: "}
                </Typography>
              </GenericCard.WorkloadTextWrapper>
              <GenericCard.WorkloadTextWrapper>
                <Typography
                  fontSize="12px"
                  fontWeight="bolder"
                  color={'#666666'}
                  fontFamily={"'Rubik', sans-serif"}
                >
                  {workload}
                </Typography>
              </GenericCard.WorkloadTextWrapper>
            </GenericCard.WorkloadText>
          </GenericCard.Workload>}
          {children}
        </GenericCard.DataBlock>
      </GenericCard.Wrapper>
    </Card>
  )

  return (
    <ThemeConsumer>
      {({ theme }) => renderComponent(theme)}
    </ThemeConsumer>
  )
}

CourseCardComponent.propTypes = {
  backgroundColor: PropTypes.string,
  borderRadius: PropTypes.string,
  children: PropTypes.node,
  companyNumber: PropTypes.number,
  fontWeightSubtitle: PropTypes.string,
  id: PropTypes.string,
  imageUrl: PropTypes.string,
  isFree: PropTypes.bool,
  isNew: PropTypes.bool,
  justifyContent: PropTypes.string,
  locations: PropTypes.array,
  onClickCard: PropTypes.func,
  subtitle: PropTypes.string,
  subtitleColor: PropTypes.string,
  title: PropTypes.string,
  titleColor: PropTypes.string,
  titleSize: PropTypes.string,
  workload: PropTypes.string
}

CourseCardComponent.defaultProps = {
  titleColor: "#391DDD", 
  titleSize: '14px'
}

export default CourseCardComponent
