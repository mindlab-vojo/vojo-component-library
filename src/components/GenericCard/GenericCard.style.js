import styled from 'styled-components'

export const Wrapper = styled.div`
  align-items: center;
  display: flex;
  position: relative;
  width: 100%;
`

export const NewTag = styled.div`
  background-color: ${props => props.theme.colors.pinkColor};
  border-radius: 5px;
  padding: 1px 6px 2px;
  position: absolute;
  right: 8px;
  top: -2px;
`

export const ImageBlock = styled.div`
  margin-right: 8px;
  margin-left: 4px;
  width: 72px;
`

export const ImageWrapper = styled.div``

export const ImageCropper = styled.div`
  display: flex;
  justify-content: center;
  width: 72px;
  height: 72px;
  position: relative;
  overflow: hidden;
  border-radius: ${props => props.borderRadius};
`

export const Image = styled.img`
  display: inline;
  margin: 0 auto;
  height: 100%;
  width: auto;
  z-index: 1;
`

export const DataBlock = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: ${props => props.justifyContent};
  min-height: 65px;
`

export const Title = styled.div`
  margin-bottom: 4px;
`

export const Subtitle = styled.div`
  margin-bottom: 4px; 
`
export const Unavailable = styled.div`
  margin: 4px 0;
`

export const Free = styled.div`
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  width: 100%;
  margin-bottom: 4px;
`

export const Workload = styled.div`
  display: flex;
  align-iteMs: center;
  width: 100%;
`

export const WorkloadText = styled.div`
  display: flex;
  align-iteMs: center;
`

export const WorkloadTextWrapper = styled.div`
    margin-left: 10px;
    &:nth-child(2) {
      margin-left: 3px;
    }
`

export const CoursesAvailable = styled.div`
  display: flex;
  align-iteMs: center;
  width: 100%;
`

export const CoursesAvailableIcon = styled.div`
  margin-right: 9px;
`
