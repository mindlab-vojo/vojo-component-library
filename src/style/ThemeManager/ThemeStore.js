import { vojo } from './../themes'

class ThemeStore {

  constructor(theme) {
    this.defaultTheme = vojo
    this.theme = theme ? theme : this.defaultTheme
  }

  configure = ({ theme }) =>  {
    if (theme) this.configureTheme(theme)
    else this.configureTheme(this.defaultTheme)
  }

  configureTheme = (theme) => {
    this.theme = theme
  }

  modify = (modifiedProps) => {
    const currentTheme = this.theme
    return { currentTheme, modifiedProps }
  }

  getTheme = () => this.theme

}

export default ThemeStore