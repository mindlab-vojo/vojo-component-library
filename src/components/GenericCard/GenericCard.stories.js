import React from 'react';
import { withKnobs, text, boolean, color } from '@storybook/addon-knobs';
import { withA11y } from '@storybook/addon-a11y';
import { action } from '@storybook/addon-actions'

import GenericCard from '.'; 
import { ThemeProvider } from "../../../src";
import categories from '../../../.storybook/categories';

export const genericCard = () => {
  const backgroundColor = color('backgroundColor', "#FFFFFF")
  const borderRadius = text('borderRadius', '0')
  const fontWeightSubtitle = text('fontWeightSubtitle', 'bolder')
  const id = text('id', 'card-id')
  const imageUrl = text('imageUrl', 'https://s3.amazonaws.com/static.vojo.com.br/prd/images/icon_servicos.PNG')
  const isNew = boolean('isNew', true)
  const isFree = boolean('isFree', true)
  const justifyContent = text('justifyContent', 'space-between')
  const onClickCard = action('onClickCard')
  const subtitle = text('subtitle', 'Descubra sites de empresas que te ajudam a gerar renda ')
  const subtitleColor = color('subtitleColor', "")
  const title = text('title', 'Entregador MÃ¡gico')
  const titleColor = color('titleColor', "#5320E8")
  const titleSize = text('titleSize', '14px')
  const workload = text('workload', '60 min')
  return (
    <GenericCard
      backgroundColor={backgroundColor}
      borderRadius={borderRadius}
      fontWeightSubtitle={fontWeightSubtitle}
      id={id}
      imageUrl={imageUrl}
      isFree={isFree}
      isNew={isNew}
      justifyContent={justifyContent}
      onClickCard={onClickCard}
      subtitle={subtitle}
      subtitleColor={subtitleColor}
      title={title}
      titleColor={titleColor}
      titleSize={titleSize}
      workload={workload}
    />
  )
}

export default {
  title: categories.Cards,
  name: 'GenericCard',
  component: genericCard,
  decorators: [
    withKnobs, 
    withA11y, 
    (story) => <ThemeProvider>{ story() }</ThemeProvider>,
    (story) => <div style={{ display: "flex" }}>{ story() }</div>,
  ],
}
  