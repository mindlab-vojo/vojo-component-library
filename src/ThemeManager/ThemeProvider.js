import React, { Component, createContext } from 'react'
import PropTypes from 'prop-types'

import ThemeStore from './ThemeStore';

// Provider and Consumer instances
const { Provider, Consumer } = createContext();

class ThemeProvider extends Component {
  static propTypes = {
    children: PropTypes.node,
    theme: PropTypes.string
  }

  state = {
    themeConfig: new ThemeStore()
  }

  componentDidMount() {
    if (this.props.theme) this.setTheme()
  }

  componentDidUpdate(prevProps) {
    if (this.props.theme !== prevProps.theme) this.setTheme()
  }

  setTheme = () => {
    const { theme } = this.props
    const { themeConfig } = this.state
    const updatedTheme = themeConfig.configure({ theme })
    this.setState({ theme: updatedTheme })
  }

  render() {
    return (
      <Provider value={{ theme: this.state.themeConfig.theme }}>
        {this.props.children}
      </Provider>
    )
  }
}

export { ThemeProvider }

export default Consumer