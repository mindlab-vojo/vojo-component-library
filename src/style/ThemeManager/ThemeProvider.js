import React, { Component, createContext } from 'react'
import PropTypes from 'prop-types'

import ThemeStore from './ThemeStore';

// Provider and Consumer instances
const { Provider, Consumer } = createContext();

class ThemeProvider extends Component {
  static propTypes = {
    children: PropTypes.node,
    theme: PropTypes.string
  }

  state = {
    themeConfig: new ThemeStore()
  }

  componentDidMount() {
    if (this.props.theme) this.setTheme()
  }

  componentDidUpdate(prevProps) {
    if (this.props.theme !== prevProps.theme) this.setTheme()
  }

  setTheme = () => {
    const { theme } = this.props
    const { themeConfig } = this.state
    const updatedTheme = themeConfig.configure({ theme })
    this.setState({ theme: updatedTheme })
  }

  getThemeFont = () => {
    const { themeConfig } = this.state
    return themeConfig.font
  }

  render() {

    const GlobalFont = this.getThemeFont()

    return (
      <Provider value={{ theme: this.state.themeConfig.theme }}>
        <GlobalFont/>
        {this.props.children}
      </Provider>
    )
  }
}

export { ThemeProvider }

export default Consumer