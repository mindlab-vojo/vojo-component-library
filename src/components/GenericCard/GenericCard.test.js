import React from 'react';
import { render } from '@testing-library/react';

import GenericCard from './';
import { ThemeProvider } from './../../style/ThemeManager/ThemeProvider';

test('Test GenericCard component render', () => {
  const component = render(
    <ThemeProvider>
      <GenericCard/>
    </ThemeProvider>
  );
  expect(component).toMatchSnapshot()
});
test('Test Reverse GenericCard component render', () => {
  const component = render(
    <ThemeProvider>
      <GenericCard isReverse/>
    </ThemeProvider>
  );
  expect(component).toMatchSnapshot()
});
