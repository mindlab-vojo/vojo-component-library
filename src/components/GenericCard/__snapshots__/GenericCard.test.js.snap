// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test GenericCard component render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-fzqBZW dcqdVe"
        id="GenericCard-undefined"
      >
        <div
          class="sc-AxjAm jDxcNw GenericCard"
        >
          <div
            class="sc-AxiKw bTpLoE CourseCard__ImageBlock"
          >
            <div
              class="sc-AxhCb ccMvAG CourseCard__ImageWrapper"
            >
              <div
                class="sc-AxhUy bZMVBu CourseCard__ImageWrapper__Cropper"
              >
                <img
                  class="sc-AxgMl TlUFs"
                />
              </div>
            </div>
          </div>
          <div
            class="sc-AxheI huQdaX"
          >
            <div
              class="sc-Axmtr eBbdId"
            >
              <p
                class="sc-fzqNJr idxwGV"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-fzqBZW dcqdVe"
      id="GenericCard-undefined"
    >
      <div
        class="sc-AxjAm jDxcNw GenericCard"
      >
        <div
          class="sc-AxiKw bTpLoE CourseCard__ImageBlock"
        >
          <div
            class="sc-AxhCb ccMvAG CourseCard__ImageWrapper"
          >
            <div
              class="sc-AxhUy bZMVBu CourseCard__ImageWrapper__Cropper"
            >
              <img
                class="sc-AxgMl TlUFs"
              />
            </div>
          </div>
        </div>
        <div
          class="sc-AxheI huQdaX"
        >
          <div
            class="sc-Axmtr eBbdId"
          >
            <p
              class="sc-fzqNJr idxwGV"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Test Reverse GenericCard component render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-fzqBZW dcqdVe"
        id="GenericCard-undefined"
      >
        <div
          class="sc-AxjAm jDxcNw GenericCard"
        >
          <div
            class="sc-AxiKw bTpLoE CourseCard__ImageBlock"
          >
            <div
              class="sc-AxhCb ccMvAG CourseCard__ImageWrapper"
            >
              <div
                class="sc-AxhUy bZMVBu CourseCard__ImageWrapper__Cropper"
              >
                <img
                  class="sc-AxgMl TlUFs"
                />
              </div>
            </div>
          </div>
          <div
            class="sc-AxheI huQdaX"
          >
            <div
              class="sc-Axmtr eBbdId"
            >
              <p
                class="sc-fzoyAV UYIYH"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-fzqBZW dcqdVe"
      id="GenericCard-undefined"
    >
      <div
        class="sc-AxjAm jDxcNw GenericCard"
      >
        <div
          class="sc-AxiKw bTpLoE CourseCard__ImageBlock"
        >
          <div
            class="sc-AxhCb ccMvAG CourseCard__ImageWrapper"
          >
            <div
              class="sc-AxhUy bZMVBu CourseCard__ImageWrapper__Cropper"
            >
              <img
                class="sc-AxgMl TlUFs"
              />
            </div>
          </div>
        </div>
        <div
          class="sc-AxheI huQdaX"
        >
          <div
            class="sc-Axmtr eBbdId"
          >
            <p
              class="sc-fzoyAV UYIYH"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
